src/Libraries/AssertNet/AssertionTypes/Extensions/CustomAttributeProviderAssertions.cs

```diff
@@ -0,0 +1,73 @@
+namespace AssertNet.AssertionTypes;
+
+/// <summary>Assertions on <see cref="ICustomAttributeProvider" />.</summary>
+public static class CustomAttributeProviderAssertions
+{
+    /// <summary>Asserts that the subject is decorated with the specified attribute.</summary>
+    /// <param name="assertion">
+    /// The value under test to assert on.
+    /// </param>
+    /// <param name="attribute">
+    /// The attribute that is expected to be amongst the custom attributes.
+    /// </param>
+    /// <param name="message">
+    /// Custom message for the assertion failure.
+    /// </param>
+    /// <typeparam name="TAssert">
+    /// The type of the assertion.
+    /// </typeparam>
+    /// <typeparam name="TSubject">
+    /// The type of the subject.
+    /// </typeparam>
+    /// <returns>The current assertion.</returns>
+    [Assertion]
+    public static TAssert IsDecoratedWith<TAssert, TSubject>(this TAssert assertion, Type attribute, string? message = null)
```

Comment:
I would expect this assertion to return a new SUT for the attribute, which would simplify the type situation